### PR TITLE
Update icon accessibility documentation

### PR DESCRIPTION
**Pull request by holdenhinkle**
## Description
Closes <https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/221>

## Testing done
New unit tests to run aXe checks against `IconHelp`, `IconSearch`, and `IconUser` components.

## Acceptance criteria
- [x] Documentation is updated to highlight when developers should use different roles
- [x] If an icon is being used alone (without support text), it should have a `role="img"` AND appropriate ARIA markup OR screen reader only descriptive text
- [x] If an icon is being used with visual (and assistive device) text, it should have `role="presentation"`
- [x] Documentation should show more explicit use cases:
  * Icon being used alone
  * Icon being used with supporting text
  * Icon being used alone inside an `<a>` tag
  * icon being used with supporting text inside an `<a>` tag
 - [x] Create and/or modify unit tests for this component to include an aXe scan

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: src/components/IconBase/IconBase.stories.jsx [hunk 1]
```diff
@@ -16,5 +16,54 @@ export const Description = () => (
     <p>
       For examples of how this is used, see the other <code>Icon</code> stories.
     </p>
+    <h2>Accessibility</h2>
```
Comment (cvalarida) [inline src/components/IconBase/IconBase.stories.jsx:19]: Let's keep this in the [design.va.gov](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/pull/543#pullrequestreview-767135897) documentation. Feel free to link to that page from here, though.

File: src/components/IconHelp/IconHelp.jsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ class IconHelp extends React.Component {
         id={id}
         pointerEvents="none"
         role={role}
+        aria-label={ariaLabel}
```
Comment (cvalarida) [inline src/components/IconHelp/IconHelp.jsx:16]: The `IconBase` prop name is `ariaLabel`, not `aria-label`, right?

File: src/components/IconSearch/IconSearch.jsx [hunk 3]
```diff
@@ -19,6 +19,7 @@ class IconSearch extends React.Component {
         id={id}
         pointerEvents="none"
         role={role}
+        aria-label={ariaLabel}
```
Comment (cvalarida) [inline src/components/IconSearch/IconSearch.jsx:22]: Same here.

**Review by cvalarida — COMMENTED** [review]
I think some of the props are off. I'd also like to see tests for conditionally rendering the `aria-label` attribute on the underlying `svg` elements. (That test would check whether the prop name mismatch is an error, too. :wink:)

File: src/components/IconBase/IconBase.stories.jsx (hunk 1 above)
Comment (holdenhinkle) [reply src/components/IconBase/IconBase.stories.jsx]: I deleted this section and linked to the icon documentation.

File: src/components/IconHelp/IconHelp.jsx (hunk 2 above)
Comment (holdenhinkle) [reply src/components/IconHelp/IconHelp.jsx:16]: That's right. We're setting the HTML aria attribute, `aria-label`, to the value of the prop `ariaLabel`.

Unless I'm missing something, I'm pretty sure this is correct.

File: src/components/IconSearch/IconSearch.jsx (hunk 3 above)
Comment (holdenhinkle) [reply src/components/IconSearch/IconSearch.jsx:22]: See comment above^

File: src/components/IconHelp/IconHelp.jsx (hunk 2 above)
Comment (holdenhinkle) [reply src/components/IconHelp/IconHelp.jsx:16]: I was writing the test and I think I just found what you were talking about:

```
    return (
      <IconBase
        className={cssClass}
        height="63"
        id={id}
        pointerEvents="none"
        role={role}
        aria-label={ariaLabel}
        viewBox="0 0 62 63"
        width="62"
      >
        <path
          fill={color}
          fillRule="evenodd"
          d="M36 50.57v-7.713c0-.375-.12-.683-.362-.924-.24-.24-.55-.362-.924-.362H27c-.375 0-.683.122-.924.363-.24.24-.362.55-.362.924v7.714c0 .376.12.684.362.926.24.24.55.36.924.36h7.714c.375 0 .683-.12.924-.36.24-.242.362-.55.362-.925zm10.286-27c0-2.356-.744-4.54-2.23-6.548a15.216 15.216 0 0 0-5.565-4.66c-2.222-1.1-4.5-1.648-6.83-1.648-6.508 0-11.477 2.853-14.906 8.558-.4.643-.294 1.206.322 1.688l5.303 4.018c.187.16.44.24.763.24.428 0 .763-.16 1.004-.48 1.42-1.823 2.572-3.055 3.456-3.698.91-.643 2.062-.964 3.455-.964 1.286 0 2.43.348 3.435 1.045S36 22.608 36 23.49c0 1.02-.268 1.836-.804 2.452-.535.616-1.446 1.22-2.732 1.808-1.687.75-3.234 1.908-4.64 3.475-1.407 1.567-2.11 3.248-2.11 5.043v1.446c0 .375.12.683.362.924.24.24.55.362.924.362h7.714c.375 0 .683-.12.924-.362.24-.24.362-.55.362-.924 0-.51.288-1.172.864-1.99a6.785 6.785 0 0 1 2.19-1.987 59.662 59.662 0 0 0 1.968-1.145c.456-.282 1.072-.75 1.85-1.407.775-.656 1.37-1.3 1.786-1.928.416-.63.79-1.44 1.125-2.43.335-.992.503-2.077.503-3.256zm15.428 7.716c0 5.598-1.38 10.76-4.138 15.49-2.76 4.726-6.502 8.47-11.23 11.228-4.728 2.76-9.89 4.14-15.49 4.14-5.597 0-10.76-1.38-15.488-4.14-4.727-2.758-8.47-6.502-11.23-11.23C1.378 42.048 0 36.885 0 31.287c0-5.6 1.38-10.76 4.138-15.49 2.76-4.727 6.503-8.47 11.23-11.23C20.096 1.81 25.258.43 30.858.43c5.597 0 10.76 1.378 15.488 4.137 4.728 2.76 8.47 6.502 11.23 11.23 2.76 4.728 4.138 9.89 4.138 15.49z"
        />
      </IconBase>
    );
```

That should be `ariaLabel={ariaLabel}`

My apologies!

**Comment by holdenhinkle** [comment]
Hi @cvalarida , I just added a bunch of new tests. I think I addressed everything you asked for. Thanks!

File: src/components/IconHelp/IconHelp.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import ReactTestUtils from 'react-dom/test-utils';
+import { expect } from 'chai';
+import { shallow, mount } from 'enzyme';
```
Comment (cvalarida) [inline src/components/IconHelp/IconHelp.unit.spec.jsx:4]: In general, we prefer new tests to be written using [`@testing-library/react`](https://testing-library.com/docs/react-testing-library/intro/). That's not written down anywhere, so I'm not gonna reject the PR for it or anything, but that library provides a lot more assurances that what we expect to happen actually is.

File: src/components/IconSearch/IconSearch.unit.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      />,
+    );
+    const IconSearchComponent = wrapper.find('IconSearch');
+    expect(IconSearchComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    const IconBaseComponent = wrapper.find('IconBase');
+    expect(IconBaseComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    wrapper.unmount();
+  });
```
Comment (cvalarida) [inline src/components/IconSearch/IconSearch.unit.spec.jsx:44]: This test is a dupe of the one below. Prefer checking the actual `svg` element.

File: src/components/IconHelp/IconHelp.unit.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      />,
+    );
+    const IconHelpComponent = wrapper.find('IconHelp');
+    expect(IconHelpComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    const IconBaseComponent = wrapper.find('IconBase');
+    expect(IconBaseComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    wrapper.unmount();
+  });
```
Comment (cvalarida) [inline src/components/IconHelp/IconHelp.unit.spec.jsx:44]: This test is a dupe of the one below. Prefer checking the actual `svg` element.

File: src/components/IconUser/IconUser.unit.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      />,
+    );
+    const IconUserComponent = wrapper.find('IconUser');
+    expect(IconUserComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    const IconBaseComponent = wrapper.find('IconBase');
+    expect(IconBaseComponent.props().ariaLabel).to.equal(ariaLabel);
+
+    wrapper.unmount();
+  });
```
Comment (cvalarida) [inline src/components/IconUser/IconUser.unit.spec.jsx:44]: This test is a dupe of the one below. Prefer checking the actual `svg` element.

**Review by cvalarida — APPROVED** [review]
Looks great! I'd prefer to see `@testing-library/react`, but I won't ask you to rewrite those tests. :sweat_smile: Only thing I'd ask is that you remove the duplicate tests. We don't so much care _how_ it renders the `aria-label` attribute (by passing the prop to `IconBase`), just that it _does._

File: src/components/IconSearch/IconSearch.unit.spec.jsx (hunk 5 above)
Comment (holdenhinkle) [reply src/components/IconSearch/IconSearch.unit.spec.jsx]: This test was removed.

File: src/components/IconHelp/IconHelp.unit.spec.jsx (hunk 6 above)
Comment (holdenhinkle) [reply src/components/IconHelp/IconHelp.unit.spec.jsx]: This test was removed.

File: src/components/IconUser/IconUser.unit.spec.jsx (hunk 7 above)
Comment (holdenhinkle) [reply src/components/IconUser/IconUser.unit.spec.jsx]: This test was removed.

File: src/components/IconHelp/IconHelp.unit.spec.jsx (hunk 4 above)
Comment (holdenhinkle) [reply src/components/IconHelp/IconHelp.unit.spec.jsx]: Good to know, thanks Chris. I'll use `@testing-library/react` going forward.

**Comment by holdenhinkle** [comment]
Thanks @cvalarida I think we're good to go now.

**Pull request merged**